### PR TITLE
Make sure to return no consumer error if we are still waiting on a leader.

### PR DESCRIPTION
**Pull request by derekcollison**
Signed-off-by: Derek Collison <derek@nats.io>

/cc @nats-io/core

### CODE REVIEW

File: server/jetstream_api.go [hunk 1]
```diff
@@ -3320,6 +3325,13 @@ func (s *Server) jsConsumerInfoRequest(sub *subscription, c *client, subject, re
 				resp.Error = jsClusterNotAvailErr
 				s.sendAPIErrResponse(ci, acc, subject, reply, string(msg), s.jsonResponse(&resp))
 			}
+			// Check here if we are a member and this is just a new consumer that does not have a leader yet.
+			if rg := ca.Group; rg.node != nil && rg.isMember(cc.meta.ID()) {
```
Comment (kozlovic) [inline server/jetstream_api.go:3329]: Is `rg` guaranteed to not be nil? I see that we check for `rg.node != nil` but not sure if `rg` itself can be nil.

File: server/jetstream_api.go [hunk 2]
```diff
@@ -3320,6 +3325,13 @@ func (s *Server) jsConsumerInfoRequest(sub *subscription, c *client, subject, re
 				resp.Error = jsClusterNotAvailErr
 				s.sendAPIErrResponse(ci, acc, subject, reply, string(msg), s.jsonResponse(&resp))
 			}
```
Comment (kozlovic) [inline server/jetstream_api.go:3327]: Do we need a `return` after sending the error in the `if js.isLeaderLess()` case?

File: server/jetstream_api.go (hunk 1 above)
Comment (derekcollison) [reply server/jetstream_api.go:3329]: Should not be but will add a check for saftey.

Comment (derekcollison) [reply server/jetstream_api.go:3329]: Will reflow.

File: server/jetstream_api.go (hunk 2 above)
Comment (derekcollison) [reply server/jetstream_api.go:3327]: Will reflow this part.

**Review by kozlovic — APPROVED** [review]
LGTM

**Pull request merged**